iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2699,46 +2762,56 @@ TEST(String100, SubstrWithDefaultSizeWorks)
 TEST(String100, SubstrWithValidPosAndSizeWorks)
 {
     ::testing::Test::RecordProperty("TEST_ID", "3cd90af2-97f4-4767-854d-d3ca726bd348");
+    constexpr uint64_t STRINGCAP = 100U;
     std::string testStdString = "Ferdinand Spitzschnueffler";
-    string<100U> testCxxString(TruncateToCapacity, testStdString);
+    string<STRINGCAP> testCxxString(TruncateToCapacity, testStdString);
 
-    std::string testStdSubstring = testStdString.substr(0, 19);
-    auto res1 = testCxxString.substr(0U, 19U);
+    constexpr uint64_t SUBSTR_POS_0 = 0;
+    constexpr uint64_t SUBSTR_COUNT_19 = 19;
+    std::string testStdSubstring = testStdString.substr(SUBSTR_POS_0, SUBSTR_COUNT_19);
+    auto res1 = testCxxString.substr(SUBSTR_POS_0, SUBSTR_COUNT_19);
     ASSERT_THAT(res1.has_value(), Eq(true));
     auto testSubstring1 = res1.value();
-    EXPECT_THAT(testSubstring1.capacity(), Eq(100U));
+    EXPECT_THAT(testSubstring1.capacity(), Eq(STRINGCAP));
     EXPECT_THAT(testSubstring1.size(), Eq(testStdSubstring.size()));
     EXPECT_THAT(testSubstring1.c_str(), StrEq(testStdSubstring));
 
-    testStdSubstring = testStdString.substr(20, 5);
-    auto res2 = testCxxString.substr(20U, 5U);
+    constexpr uint64_t SUBSTR_POS_20 = 20;
+    constexpr uint64_t SUBSTR_COUNT_5 = 5;
+    testStdSubstring = testStdString.substr(SUBSTR_POS_20, SUBSTR_COUNT_5);
+    auto res2 = testCxxString.substr(SUBSTR_POS_20, SUBSTR_COUNT_5);
     EXPECT_THAT(res2.has_value(), Eq(true));
     auto testSubstring2 = res2.value();
-    EXPECT_THAT(testSubstring2.capacity(), Eq(100U));
+    EXPECT_THAT(testSubstring2.capacity(), Eq(STRINGCAP));
     EXPECT_THAT(testSubstring2.size(), Eq(testStdSubstring.size()));
     EXPECT_THAT(testSubstring2.c_str(), StrEq(testStdSubstring));
 
-    testStdSubstring = testStdString.substr(0, 26);
-    auto res3 = testCxxString.substr(0U, 26U);
+    constexpr uint64_t SUBSTR_COUNT_26 = 26;
+    testStdSubstring = testStdString.substr(SUBSTR_POS_0, SUBSTR_COUNT_26);
+    auto res3 = testCxxString.substr(SUBSTR_POS_0, SUBSTR_COUNT_26);
```

Comment:
This also hurts readability. Much noise without much information. For future fixes I would suggest to use just `POS` and `COUNT` or `POS_A`/`POS_B`. There is not much value to just repeat the number. If it is part of the name, I would rather use the magic number instead of creating a constexpr.

I just had the idea to create a header with
```cpp
constexpr uint64_t _0 {0};
constexpr uint64_t _1 {1};
constexpr uint64_t _2 {2};
constexpr uint64_t _3 {3};
constexpr uint64_t _4 {4};
...
```
and use those constants all over the place :laughing: